docs/mozilla-traffic-cop.rst

```diff
@@ -0,0 +1,104 @@
+.. This Source Code Form is subject to the terms of the Mozilla Public
+.. License, v. 2.0. If a copy of the MPL was not distributed with this
+.. file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+.. _mozillatrafficcop:
+
+===================
+Mozilla Traffic Cop
+===================
+
+`mozilla-traffic-cop.js` handles redirecting users to different A/B/x variations
+through a query parameter.
+
+**`mozilla-traffic-cop.js` requires `mozilla-cookie-helper.js`.**
+
+How It Works
+------------
+
+After verifying the configuration, the Traffic Cop first makes sure the user is
+not currently viewing a variation (to avoid a redirect loop). It then chooses a
+random percentage (1-100).
+
+If the percentage falls within the total percentage targeted by the config, it
+will choose the appropriate variation and redirect the user.
+
+If the percentage exceeds that of the config, no redirect will occur. (This
+could be considered the "control".)
+
+If a variation is chosen, a cookie is set (for the session only) that will
+send the user back to the same variation if/when the page is again visited.
+
+Any querystring parameters present when a user initially lands on a page will be
+propagated to the variation redirect.
+
+Configuration
+-------------
+
+Each instance of the Traffic Cop requires two pieces of configuration:
+
+- A string ID that is unique to other currently running tests (to avoid
+confusion when reading cookies)
+- A variations object that lists all variations by querystring along with the
+associated visitor percentage
+
+An implementation might look like::
+
+    var cop = new Mozilla.TrafficCop({
+        id: 'exp_firefox_new_headline',
+        variations: {
+            'v=1': 25,
+            'v=2': 25,
+            'v=3': 25
+        }
+    });
+
+    cop.init();
+
+In the above example, the string `exp_firefox_new_headline` will be used as the
+cookie name should a variation be chosen.
+
+The test will have 3 variations, each targeting 25% of users.
+
+How a Variation is Chosen
+-------------------------
+
+Variations are sorted in the order provided, and percentages are tallied to
+create tiers. Take the following config::
+
+    var cop = new Mozilla.TrafficCop({
+        id: 'exp_firefox_new_headline',
+        variations: {
+            'v=c': 25,
+            'v=a': 25,
+            'v=f': 25
+        }
+    });
+
+The implied tiers would be:
+
+- v=c: 1-25
+- v=a: 26-50
+- v=f: 51-75
+- (no redirect): 76-100
+
+So, if the random percentage chosen was 44, the user would be redirected to
+`v=a`.
+
+Implementation
+--------------
+
+To add a Traffic Cop to a page, create a new JS bundle for the experiment that
+includes:
+
+- `mozilla-cookie-helper.js`
+- `mozilla-traffic-cop.js`
+- A new JS file that configures and initializes the experiment
+
+Place this new bundle in the `extrahead` block of the page
```

Comment:
Should we also intentionally not run the lib in unsupported browsers like IE < 9? We could use a conditional comment or something similar. Seems like it could improve the reliability of test data if we avoid putting users who get no CSS and minimal JS into the testing pots.